apps/src/templates/progress/GoogleClassroomShareButton.jsx

```diff
@@ -0,0 +1,103 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import Button from '../Button';
+import i18n from '@cdo/locale';
+
+// https://developers.google.com/classroom/brand
+const styles = {
+  label: {
+    paddingLeft: 16,
+    fontFamily: 'Roboto',
+    fontWeight: 700
+  },
+  container: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'start'
+  }
+};
+class GoogleClassroomShareButton extends React.Component {
+  static propTypes = {
+    buttonId: PropTypes.string.isRequired,
+    url: PropTypes.string.isRequired,
+    itemtype: PropTypes.string.isRequired,
+    title: PropTypes.string,
+    height: PropTypes.number,
+    courseid: PropTypes.number
+  };
+
+  constructor(props) {
+    super(props);
+    this.loadApi = this.loadApi.bind(this);
+    this.waitForGapi = this.waitForGapi.bind(this);
+    this.renderButton = this.renderButton.bind(this);
+  }
+
+  componentDidMount() {
+    if (this.gapiReady()) {
+      this.renderButton();
+    } else {
+      this.loadApi();
+    }
+  }
+
+  /* The Google Classroom Share Button is only available through their js api,
+   * so we have to add it to the page on load. Each instance of this
+   * component will wait until the api is ready before rendering the button,
+   * but only the first will kick of the loading process.
+   */
+  loadApi() {
+    if (!document.getElementById('gapi')) {
+      window.___gcfg = {
+        parsetags: 'explicit'
+      };
+
+      const gapi = document.createElement('script');
+      gapi.src = 'https://apis.google.com/js/platform.js';
```

Comment:
Will we need to add apis.google.com to our list of "Sites to Unblock" in the [technical requirements](https://support.code.org/hc/en-us/articles/202591743-Technical-requirements-for-Code-org) and [it](https://code.org/educate/it) page?

Reply:
nice catch! i didn't know about those pages. do you know who manages them?

Reply:
That's Josh Schulte and Marina. You're also welcome to update them yourself anytime and just make sure you get their signoff. I'm actually not sure if we need this one on the allow-list. I don't see other similar host names (but that could just be an accident). Josh and Marina would know. 

To update: The IT page lives here: https://github.com/code-dot-org/code-dot-org/blob/staging/pegasus/sites.v3/code.org/public/educate/it.md
The technical requirements page is a blog post and instructions for updating are here and the following few threads: https://codedotorg.slack.com/archives/C039X1ZLX/p1592952987042400